cmd/device-hub-cli/helpers.go

```diff
@@ -108,9 +111,66 @@ func dial() (*grpc.ClientConn, proto.HubClient, error) {
 	return conn, proto.NewHubClient(conn), nil
 }
 
-type roundTripFunc func(cli proto.HubClient, in iocodec.Decoder, out iocodec.Encoder) error
+type rawConf []byte
+
+func (r rawConf) Decode(target interface{}) error {
+	err := yaml.Unmarshal(r, target)
+	if err != nil {
+		return fmt.Errorf("error decoding data: %s", err.Error())
+	}
+	return nil
+}
+
+// Represent a conf file, Data is basically used to order a cliConf Slice
+// Raw contains the file content
+type cliConf struct {
+	Data map[string]interface{}
+	Raw  rawConf
+}
+
+// Load the configuration file to Data
+func (c *cliConf) Load(filePath string) (err error) {
+
+	c.Raw, err = ioutil.ReadFile(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to read file [%s]: %s", filePath, err.Error())
+	}
+	err = yaml.Unmarshal(c.Raw, &c.Data)
+	if err != nil {
+		return fmt.Errorf("error parsing file [%s]: %s", filePath, err.Error())
+	}
+	return nil
+}
+
+type cliConfSlice struct {
```

Comment:
I am wondering if you could restructure the code to on each command return a cliConfSlice, order as per the intention of the command, validate, send over the wire?